### PR TITLE
Allow zigbee image generation to use min_max_hw_ver

### PR DESCRIPTION
**Pull request by sebastiandraus**
Add optional arguments to allow generate Zigbee OTA Image with specified minimum and maximum hardware version.

### CODE REVIEW

File: nordicsemi/zigbee/ota_file.py [hunk 1]
```diff
@@ -74,25 +78,45 @@ def __init__(self,
...
+        if (type(min_hw_version) is int) and (type(max_hw_version) is int):
+            # Calculate additional length of minimum and maximum hardware version fields
+            ota_min_max_hw_length = OTA_UPGRADE_MIN_HW_VERSION_LENGTH + OTA_UPGRADE_MAX_HW_VERSION_LENGTH
+            # Add additional bytes to formating string
+            ota_header_pack_format += 'HH'
+            # Add min and max hardware version to ota header arguments
+            ota_header_pack_args.append(min_hw_version)
+            ota_header_pack_args.append(max_hw_version)
+            # Set bit "Hardware Versions Present" in OTA Header Field Control
+            ota_header_pack_args[3] += OTA_UPGRADE_FIELD_CONTROL_WITH_HW_VER
```
Comment (wbober) [inline nordicsemi/zigbee/ota_file.py]: I'd define constants for header field positions, e.g., ```OTA_UPGRADE_FILE_HEADER_LENGTH_FIELD = 3```, or something like that.

File: nordicsemi/__main__.py [hunk 2]
```diff
@@ -844,6 +854,14 @@ def generate(zipfile,
     if zigbee:
         inner_external_app = False
 
+    # Warn user if minimal/maximum zigbee ota hardware version are not correct:
+    #   * only one of them is given
+    #   * minimum version is higher than maximum version
+    if (type(zigbee_ota_min_hw_version) is not int) != (type(zigbee_ota_max_hw_version) is not int):
+        click.echo('Warning: min/max zigbee ota hardware version is missing. Discarding min/max hardware version.')
```
Comment (tomchy) [inline nordicsemi/__main__.py]: Personally I prefer positive checks against `None` instead of `not int`.

File: nordicsemi/__main__.py [hunk 3]
```diff
@@ -844,6 +854,14 @@ def generate(zipfile,
...
+    elif (type(zigbee_ota_min_hw_version) is int) and (zigbee_ota_min_hw_version > zigbee_ota_max_hw_version):
+        click.echo('Warning: zigbee-ota-min-hw-version is higher than zigbee-ota-max-hw-version.')
+
```
Comment (tomchy) [inline nordicsemi/__main__.py]: Please add an assert that `zigbee_ota_hw_version` is between max and min.

File: nordicsemi/zigbee/ota_file.py [hunk 4]
```diff
@@ -74,25 +78,45 @@ def __init__(self,
...
+                                OTA_UPGRADE_FIELD_CONTROL,
+                                manufacturer_code,
+                                image_type,
+                                file_version,
+                                OTA_UPGRADE_FILE_HEADER_STACK_PRO,
+                                chr(len(comment)),
+                                bytes(comment.encode('ascii')),
+                                total_len]
+
+        if (type(min_hw_version) is int) and (type(max_hw_version) is int):
```
Comment (tomchy) [inline nordicsemi/zigbee/ota_file.py]: If you choose to check for `None` - this line should be aligned too.

File: nordicsemi/zigbee/ota_file.py [hunk 5]
```diff
@@ -74,25 +78,45 @@ def __init__(self,
...
+            # Calculate additional length of minimum and maximum hardware version fields
+            ota_min_max_hw_length = OTA_UPGRADE_MIN_HW_VERSION_LENGTH + OTA_UPGRADE_MAX_HW_VERSION_LENGTH
+            # Add additional bytes to formating string
+            ota_header_pack_format += 'HH'
```
Comment (tomchy) [inline nordicsemi/zigbee/ota_file.py]: Is there any check for values passed as min/max, so each of them fit in 8-bit field?

File: nordicsemi/__main__.py [hunk 6]
```diff
@@ -609,6 +609,14 @@ def pkg():
               help='The zigbee OTA fw version.',
               required=False,
               type=BASED_INT_OR_NONE)
+@click.option('--zigbee-ota-min-hw-version',
+              help='The zigbee OTA minimum hw version.',
+              required=False,
+              type=BASED_INT_OR_NONE)
+@click.option('--zigbee-ota-max-hw-version',
+              help='The zigbee OTA maximum hw version.',
```
Comment (tomchy) [inline nordicsemi/__main__.py]: As there are so many hardware versions available, please use `OTA client` inside help message.

File: nordicsemi/zigbee/ota_file.py [hunk 7]
```diff
@@ -48,6 +48,10 @@
 OTA_UPGRADE_SUBELEMENT_HEADER_SIZE    = 2 + 4  # Tag ID + Length Field
 OTA_UPGRADE_SUBELEMENT_TRIGGER_TYPE   = 0xCDEF
 OTA_UPGRADE_SUBELEMENT_TRIGGER_LENGTH = 1 + 4 + 4 + 4 + 4 # See background_dfu_trigger_t in the nRF5 SDK for explanation
+
+OTA_UPGRADE_FIELD_CONTROL_WITH_HW_VER = (1 << 2)
```
Comment (tomchy) [inline nordicsemi/zigbee/ota_file.py]: Just a thought: what do you think about placing "bit mask" inside the name of this constant?

**Pull request merged**